test/demo/resolveByIntent.inttest.ts

```diff
@@ -1,53 +1,178 @@
+                        expect(actualAppNames.sort()).toEqual(expectedAppNames);
+                    });
+                });
+            });
+        });
+
+        describe('With an ad-hoc app running', () => {
+            beforeEach(async () => {
+                setupStartNonDirectoryApp(testAppNotInDirectory);
+            });
+
+            describe('But the ad-hoc app has not registered a listener for the intent', () => {
+                test('When calling findIntent with the intent, the ad-hoc app is NOT included in the results', async () => {
+                    const expectedAppNames = getDirectoryAppsForIntent(directory, validIntent);
+
+                    // Resolve the valid intent
+                    const actualAppNames = await findIntent(validIntent).then(apps => apps.map(app => app.name));
+
+                    // Compare the two arrays (sorted as order doesn't matter)
+                    expect(actualAppNames.sort()).toEqual(expectedAppNames);
+                });
+            });
+            describe('And the ad-hoc app has registered a listener for the intent', () => {
+                beforeEach(async () => {
+                    await fdc3Remote.addIntentListener(testAppNotInDirectory, validIntent);
+                    await delay(300);
+                });
+
+                test('When calling findIntent with the intent, the ad-hoc app is returned among the results', async () => {
+                    const directoryAppsForIntent = getDirectoryAppsForIntent(directory, validIntent);
+                    const expectedAppNames = directoryAppsForIntent.concat([testAppNotInDirectory.name]).sort();
+
+                    // Resolve the valid intent
+                    const actualAppNames = await findIntent(validIntent).then(apps => apps.map(app => app.name));
+
+                    // Compare the two arrays (sorted as order doesn't matter)
+                    expect(actualAppNames.sort()).toEqual(expectedAppNames);
+                });
+            });
         });
     });
 
     describe('With context', () => {
         test.todo('TBD what resolving intent with context means as the spec is unclear');
     });
 });
+
+function getDirectoryAppsForIntent(directory: DirectoryApp[], targetIntent: string) {
+    return directory.filter(app => app.intents && app.intents.some(intent => intent.name === targetIntent)).map(app => app.name).sort();
+}
+
+function findIntent(intent: string, context: Context | undefined = undefined, sendFromIdentity?: Identity): Promise<DirectoryApp[]> {
+    return fdc3Remote.findIntent(sendFromIdentity || testManagerIdentity, intent, context)
+        .then(appIntent => appIntent.apps);
+}
+
+// TODO: The below helper functions and types are copied from `raiseIntent` tests. Should put everything in a util file
+
+function setupOpenDirectoryApp(app: AppIdentity) {
+    beforeEach(async () => {
+        await fdc3Remote.open(testManagerIdentity, app.uuid);
```

Comment:
Our in open.inttest we're doing open(testManagerIdentity, app.name). Suggest do same for consistency